### PR TITLE
Building A Personality Test App with Java

### PR DESCRIPTION
**Pull request by CaptainBKola**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

-[ ]@ahmadmardeni1 @hectorkambow @RuthAMare The duplicate to the article previously closed by    ###@hectorkambow.

### CODE REVIEW

**Comment by CaptainBKola** [comment]
@ahmadmardeni1 @hectorkambow @RuthAMare.  duplicate to the article previously closed.

File: content/articles/myers-briggs-personality-test-app/index.md [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+// is the same as
+int oneDarray [], numbers[][];
+```
+### How to create arrays
+
+We create arrays with the new keyword or with array literals. The index of the array is a fixed size. Each element has its respective default values.
+
+#### Creating an array with the new keyword
+
+We create an empty array like this;
```
Comment (RuthAMare) [inline content/articles/myers-briggs-personality-test-app/index.md:84]:
```suggestion
We create an empty array like this:
```

File: content/articles/myers-briggs-personality-test-app/index.md [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+
+```java
+String [] arrayOfStrings= new int [4] // array of Strings;
+```
+#### Creating an array with array literals
+
+```java
+String [ ] subjects = {“English”,”Mathematics”};
+```
+Whenever we declare and initialize an array with array literals, we wouldn't need the new keyword except for a boolean data type. We declare and initialize boolean values like this;
```
Comment (RuthAMare) [inline content/articles/myers-briggs-personality-test-app/index.md:115]:
```suggestion
Whenever we declare and initialize an array with array literals, we wouldn't need the new keyword except for a boolean data type. We declare and initialize boolean values like this:
```

File: content/articles/myers-briggs-personality-test-app/index.md [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+String [] oneDstringValue ={“room”, ”door”}
+oneDstringValue [1] = “stores”;
+```
+After the brief introduction to some basics of java array, let's dive into building our app.
+
+### Building the personality test app
+
+#### Step one: Create a file with the name PersonalityTest.java
+
+In this file, define a class with the same name PersonalityTest like this;
```
Comment (RuthAMare) [inline content/articles/myers-briggs-personality-test-app/index.md:132]:
```suggestion
In this file, define a class with the same name PersonalityTest like this:
```

File: content/articles/myers-briggs-personality-test-app/index.md [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+---
+If you are trying to build a personality test app or a quiz app, this article is for you. This article will use the concept of Java arrays, StringBuilder class, exception handling, and format specifiers in Java to make a personality test app.
+<!--more-->
+### Table of contents
+
+- [How the Personality Test Works](#how-the-personality-test-works)
+- [Prerequisites](#prerequisites)
+- [Introduction to Java Arrays](#introduction-to-java-arrays)
+- [Building the Personality Test App](#building-the-personality-test-app)
+- [Explaining the Methods](#explaining-the-methods)
```
Comment (RuthAMare) [inline content/articles/myers-briggs-personality-test-app/index.md:25]:
```suggestion
- [Explaining the methods](#explaining-the-methods)
```

File: content/articles/myers-briggs-personality-test-app/index.md [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+ - url: /engineering-education/building-a-personality-test-app-with-java/hero.jpeg
+   alt: Personality test hero image
+---
+If you are trying to build a personality test app or a quiz app, this article is for you. This article will use the concept of Java arrays, StringBuilder class, exception handling, and format specifiers in Java to make a personality test app.
+<!--more-->
+### Table of contents
+
+- [How the Personality Test Works](#how-the-personality-test-works)
+- [Prerequisites](#prerequisites)
+- [Introduction to Java Arrays](#introduction-to-java-arrays)
```
Comment (RuthAMare) [inline content/articles/myers-briggs-personality-test-app/index.md:23]:
```suggestion
- [Introduction to Java arrays](#introduction-to-java-arrays)
- [How to create arrays](#how-to-create-arrays)
```

File: content/articles/myers-briggs-personality-test-app/index.md [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+          }else {
+              throw new IllegalArgumentException("Wrong choice; choose A or B");
+          }
+      }catch (IllegalArgumentException ex){
+         System.err.println(ex.getMessage());
+   }
+}
+}
+```
+#### This method serves questions and gets options
```
Comment (RuthAMare) [inline content/articles/myers-briggs-personality-test-app/index.md:350]:
```suggestion
#### Method for serving questions and getting options
```

**Review by RuthAMare — CHANGES_REQUESTED** [review]
Hey @CaptainBKola .
Thanks for your determination to polish this article.
See the changes below and commit them when you're done incorporating them.
Also, space the work adequately especially the steps under building the personality test app. Leave at least a blank line before the next step.
Also, rename the branch on which you have opened this PR to building-a-personality-test-app-with-java instead of new-profile.

Let me know when you're done.

**Comment by CaptainBKola** [comment]
everything is fine now @RuthAMare

**Comment by CaptainBKola** [comment]
@RuthAMare  renaming a branch is while on pr is not technically feasible on github.

**Comment by CaptainBKola** [comment]
@RuthAMare renaming a branch while on pr is not technically feasible on github.

**Comment by CaptainBKola** [comment]
@RuthAMare, I can only close this pr and make new pr with a new branch and the old one deleted

**Comment by CaptainBKola** [comment]
#Duplicate pull request

**Comment by CaptainBKola** [comment]
Duplicate pull request

**Comment by RuthAMare** [comment]
@CaptainBKola please add the title ### How to create arrays to the table of contents
It's part of the content but not listed on the table of contents

File: content/articles/myers-briggs-personality-test-app/index.md (hunk 5 above)
Comment (RuthAMare) [reply content/articles/myers-briggs-personality-test-app/index.md]: The above is what I want you to add

**Comment by RuthAMare** [comment]
Hey @ahmadmardeni1, @CaptainBKola seems to have fixed most of the issues apart from the above.

**Comment by CaptainBKola** [comment]
Oh okay @RuthAMare thank you

**Comment by RuthAMare** [comment]
Good Job @CaptainBKola.

**Comment by CaptainBKola** [comment]
@RuthAMare thank you

**Comment by hectorkambow** [comment]
👍 
Overall nice work @CaptainBKola 

cc @RuthAMare

**Comment by CaptainBKola** [comment]
@hectorkambow @RuthAMare @ahmadmardeni1  thank you guys

**Pull request merged**